swift/example_code/s3/basics/Tests/basics-tests/basics-tests.swift

```diff
@@ -0,0 +1,649 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+import XCTest
+import class Foundation.Bundle
+import Dispatch
+import ClientRuntime
+import Fakery
+@testable import ServiceHandler
+
+let DEBUG_SEM = false
+struct TestWaiter {
+    let sem: DispatchSemaphore
+    var name: String = "Unnamed"
+    var timeout: Double = 2.0
+
+    init() {
+        sem = DispatchSemaphore(value: 0)
+        if DEBUG_SEM == true {
+            print("Sem init: \(name)")
+        }
+    }
+
+    init(name: String, timeout: Double = 2.0) {
+        self.name = name
+        self.timeout = timeout
+
+        sem = DispatchSemaphore(value: 0)
+        if DEBUG_SEM == true {
+            print("Sem init: \(name)")
+        }
+    }
+
+    /// Signal the DispatchSemaphore, ending the lock.
+    public func signal() {
+        if DEBUG_SEM == true {
+            print("Sem signal: \(name)")
+        }
+        sem.signal()
+    }
+
+    /// Wait until the semaphore is unlocked, or until
+    /// the timeout period has elapsed, whichever comes
+    /// first.
+    ///
+    /// - Returns: true if the timeout elapsed; false if
+    ///            the semaphore was signaled. Ignorable.
+    @discardableResult public func wait() -> Bool {
+        var timedOut: Bool = false
+
+        if DEBUG_SEM == true {
+            print("Sem wait: \(name) start")
+        }
+        
+        let timeoutResult = sem.wait(timeout: .now() + self.timeout)
+        if (timeoutResult == .timedOut) {
+            timedOut = true
+        }
+
+        if DEBUG_SEM == true {
+            print("Sem wait: \(name) complete")
+        }
+        return timedOut
+    }
+}
+
+/// Returns a unique filename for use in testing. The filename
+/// is just a UUID, optionally with a period and an extension
+/// added. The returned name is invalid for S3 use if isValid is
+/// true.
```

Comment:
Returns a unique file name for testing. The file name is a UUID, optionally with a period and extension added. If isValid is true, the returned name can't be used for Amazon S3.